### PR TITLE
Add error message when GPU is not available

### PR DESCRIPTION
**Pull request by szalpal**
## Category:

**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
Currently, when DALI pipeline is created in Triton, but user forgets to pass `--gpus` flag to the run command, he gets an obscure error message:
```
dlopen libcuda.so failed!. Please install GPU dirverTraceback (most recent call last):
  File "<string>", line 8, in <module>
  File "/opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/_utils/autoserialize.py", line 77, in invoke_autoserialize
    dali_pipeline().serialize(filename=filename)
  File "/opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/pipeline.py", line 1261, in serialize
    self._init_pipeline_backend()
  File "/opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/pipeline.py", line 725, in _init_pipeline_backend
    self._pipe = b.Pipeline(self._max_batch_size,
RuntimeError: [/opt/dali/dali/core/device_guard.cc:31] Assert on "cuInitChecked()" failed: Failed to load libcuda.so. Check your library paths and if the driver is installed correctly.
Stacktrace (31 entries):
[frame 0]: /opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/libdali_core.so(+0x233fb) [0x7fed69bb13fb]
[frame 1]: /opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/libdali_core.so(dali::DeviceGuard::DeviceGuard(int)+0x1a8) [0x7fed69bd4548]
[frame 2]: /opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/libdali.so(dali::Pipeline::Init(int, int, int, long, bool, bool, bool, unsigned long, bool, int, int, dali::QueueSizes)+0x50) [0x7fed6f81a620]
[frame 3]: /opt/tritonserver/backends/dali/conda/envs/dalienv/lib/python3.10/site-packages/nvidia/dali/backend_impl.cpython-310-x86_64-linux-gnu.so(dali::Pipeline::Pipeline(int, int, int, long, bool, int, bool, unsigned long, bool, int, int)+0x363) [0x7fed64c05a53]
```
This PR introduces more descriptive error message.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by szalpal** [comment]
!build

File: dali/pipeline/pipeline.cc [hunk 1]
```diff
@@ -180,6 +180,11 @@ void Pipeline::Init(int max_batch_size, int num_threads, int device_id, int64_t
                     bool pipelined_execution, bool separated_execution, bool async_execution,
                     size_t bytes_per_sample_hint, bool set_affinity, int max_num_stream,
                     int default_cuda_stream_priority, QueueSizes prefetch_queue_depth) {
+    DALI_ENFORCE(cuInitChecked() || device_id == CPU_ONLY_DEVICE_ID,
+                "You are trying to create a GPU DALI pipeline, while CUDA is not available. "
+                "Please install CUDA or set `device_id = None` in Pipeline constructor. "
+                "If running inside Docker container, you may need to pass `--gpus all` option.");
```
Comment (JanuszL) [inline dali/pipeline/pipeline.cc:186]:
```suggestion
                "If running inside Docker container, you may need to use  `--gpus` option.");
```

**Review by stiepan — APPROVED** [review]

**Review by stiepan — COMMENTED** [review]
I recall we wanted to be careful not to call cuda-related API with device_id=-1.

**Comment by szalpal** [comment]
!build

**Comment by szalpal** [comment]
!build

**Review by stiepan — APPROVED** [review]

**Review by banasraf — APPROVED** [review]

**Pull request merged**